sql/src/main/java/io/crate/planner/operators/HashJoin.java

```diff
@@ -128,14 +128,20 @@ public ExecutionPlan build(PlannerContext plannerContext,
             extractHashJoinSymbolsFromJoinSymbolsAndSplitPerSide(tablesSwitched);
 
         ResultDescription leftResultDesc = leftExecutionPlan.resultDescription();
+        ResultDescription rightResultDesc = rightExecutionPlan.resultDescription();
         Collection<String> joinExecutionNodes = leftResultDesc.nodeIds();
 
         List<Symbol> leftOutputs = leftLogicalPlan.outputs();
         List<Symbol> rightOutputs = rightLogicalPlan.outputs();
         MergePhase leftMerge = null;
         MergePhase rightMerge = null;
 
-        ResultDescription rightResultDesc = rightExecutionPlan.resultDescription();
+        // we can only run the join distributed if no remaining limit or offset must be applied on the source relations.
+        // because on distributed joins, every join is running on a slice (modulo) set of the data and so no limit/offset
+        // could be applied.
```

Comment:
A suggestion: maybe add: `...applied. Limit/offset can only be applied on the whole data set after all partial rows from the shards are merged`